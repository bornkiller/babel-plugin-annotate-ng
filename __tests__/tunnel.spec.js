/**
 * @description - babel-plugin-annotate-ng tunnel none-comment unit test
 * @author - bornkiller <hjj491229492@hotmail.com>
 */

const path = require('path');
const babel = require('babel-core');
const annotate = require('../src');
// fixture
const babelOptions = {
  presets: [],
  plugins: [annotate],
  babelrc: false
};

describe('babel-plugin-annotate-ng Tunnel', function () {
  it('FunctionDeclaration', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Tunnel/find-fruit.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });

  it('ClassDeclaration', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Tunnel/polaris.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });

  it('VariableDeclaration', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Tunnel/warrior.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });
});
