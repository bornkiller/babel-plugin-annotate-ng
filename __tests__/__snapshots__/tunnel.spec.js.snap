// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-annotate-ng Tunnel Class Declaration 1`] = `
"export class HelloWorld {
  constructor() {}

  ngOnInit() {}

  render() {}
}"
`;

exports[`babel-plugin-annotate-ng Tunnel Function Declaration 1`] = `
"export function find(fruit) {
  const inventory = [{ name: 'apples', quantity: 2 }, { name: 'bananas', quantity: 0 }, { name: 'cherries', quantity: 5 }];

  return inventory.find(inventory => inventory.name === fruit);
}"
`;
